network/p2p/middleware.go

```diff
@@ -392,17 +406,31 @@ func (m *Middleware) handleIncomingStream(s libp2pnetwork.Stream) {
 	}
 	_, isStaked := m.ov.Identities().ByNodeID(nodeID)
 
+	ctx, cancel := context.WithTimeout(m.ctx, LargeMsgUnicastTimeout)
+
+	deadline, _ := ctx.Deadline()
+	err = s.SetReadDeadline(deadline)
+	if err != nil {
+		log.Err(err).Msg("failed to set read deadline for stream")
```

Comment:
is logging here sufficient? not other action needed? I guess it would just default to the same behaviour as if we didn't try to set a deadline?

Reply:
So, there are two options here.

We could either say, if we fail to set the deadline, then we abort and close the stream immediately.

Otherwise, we could just fallback to the default behavior, which means potentially that the read / write could block forever.

I'm happy to go with either approach, but in the end I think in practice we will never fail to set the deadline. None of the implementations ever returns an error: https://github.com/libp2p/go-yamux/blob/cd22a37b789cf6e1fe4f2c2cff6da5df7a49dfac/stream.go#L457-L486